master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -227,13 +227,13 @@ private[celeborn] class Master(
   masterSource.addGauge(MasterSource.REGISTERED_SHUFFLE_COUNT) { () =>
     statusSystem.registeredShuffleCount
   }
-  masterSource.addGauge(MasterSource.WORKER_COUNT) { () => statusSystem.workers.size }
+  masterSource.addGauge(MasterSource.WORKER_COUNT) { () => statusSystem.getWorkers.size }
   masterSource.addGauge(MasterSource.LOST_WORKER_COUNT) { () => statusSystem.lostWorkers.size }
   masterSource.addGauge(MasterSource.EXCLUDED_WORKER_COUNT) { () =>
     statusSystem.excludedWorkers.size + statusSystem.manuallyExcludedWorkers.size
   }
   masterSource.addGauge(MasterSource.AVAILABLE_WORKER_COUNT) { () =>
-    statusSystem.workers.asScala.count { w =>
+    statusSystem.getWorkers.asScala.count { w =>
```

Comment:
Why can’t we use workersMap.values directly? If we use a copy for workers, which may introduce unnecessary overhead.

Reply:
It seems acceptable.

Reply:
UnmodifiableCollection will not copy the elements, it just limit the add/remove/putAll method access.

Reply:
I just set the workersMap to private to prevent that there is typo to get the worker by WorkerInfo instead of WorkerId.

Reply:
> UnmodifiableCollection will not copy the elements, it just limit the add/remove/putAll method access.

I see, thanks!

Reply:
> I just set the workersMap to private to prevent that there is typo to get the worker by WorkerInfo instead of WorkerId.

All maps have this problem because get or contains will not verify its key type. Just expose the workersmap may be OK.

Reply:
ok